hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/RSProcedureDispatcher.java

```diff
@@ -233,13 +233,19 @@ public void run() {
 
     private int numberOfAttemptsSoFar = 0;
     private long maxWaitTime = -1;
+    private final long rsRpcRetryInterval;
```

Comment:
This is not related to the title here?

Reply:
@Apache9 This is useful for the cases where there is an injected environment edge and when the first RS RPC request fails causing it to go to the retry block.

In the absence of this , the default timeout would be set to 100 ms and the DelayedUtil class is meant to execute the retry after 100 ms. However as per the getRemainingTime() logic here  (https://github.com/apache/hbase/blob/5b01e613fbbb92e243e99a1d199b4ffbb21ed2d9/hbase-procedure/src/main/java/org/apache/hadoop/hbase/procedure2/util/DelayedUtil.java#L94), the equation would be evaluated to 

EnvironmentEdgeManager.currentTime() >= EnvironmentEdgeManager.currentTime() + 100 which would never get evaluated to true in case of injected edge and retry never happens. Hence this config helps you to override it to 0 in cases we want to test with a manual injected environment edge

Reply:
An example would be Master trying to open the meta region even before the RS is online, In this case the retry of opening the meta region would never happen in the case of injected environment edge and master would never finish the initialization

Reply:
Agree, this doesn't fit the scope implied by the JIRA and PR description. Remove it here, submit as new issue/PR?

Reply:
Sure @apurtell , created a new JIRA to track this : https://issues.apache.org/jira/browse/HBASE-24511

Will close this PR for now and send two follow up PRs for the newly created PR followed by the PR for this JIRA